### ISSUE TITLE
27kpng_model_best.pth.tar The Compressed data has been corrupted 

### ISSUE THREAD

**Issue by Tristacheng** [vinthony/deep-blind-watermark-removal#5]
Hi, thank you for sharing your work.
I want to run your network on my data, But when I decompressed 27kpng_model_best.pth.tar, it says the file is corrupted, could you   please upload it again？

**Comment by vinthony** [comment]
Hi, you do not need to decompress the model.

just load the model to the neural network via:

```
... = torch.load(/path/to/27kpng_model_best.pth.tar)
```

you can find the whole framework via [our online demo](https://colab.research.google.com/drive/1pYY7byBjM-7aFIWk8HcF9nK_s6pqGwww?usp=sharing).

**Comment by Tristacheng** [comment]
> Hi, you do not need to decompress the model.
> 
> just load the model to the neural network via:
> 
> ```
> ... = torch.load(/path/to/27kpng_model_best.pth.tar)
> ```
> 
> you can find the whole framework via [our online demo](https://colab.research.google.com/drive/1pYY7byBjM-7aFIWk8HcF9nK_s6pqGwww?usp=sharing).

thank you for your reply, it's ok

**Comment by Tristacheng** [comment]
hi, when I use this model on my data, the results are not pleasing.
could you give me some advice?
@vinthony 
![捕获2](https://user-images.githubusercontent.com/12381965/115197460-96bf9f80-a123-11eb-8402-c4d14c10be4d.JPG)
![捕获](https://user-images.githubusercontent.com/12381965/115197474-9b845380-a123-11eb-913e-6d6a81fa5783.JPG)

**Comment by vinthony** [comment]
Hi, removing the watermark total blindly is still very challenging. In our experiments, the current model is trained on the synthesized datasets( only 27k as described in the paper) which might just follow the specific distribution.

For the images in the wild, there are many challenges that might influence the performance of the models. For example, the released model is only trained on colorful watermarks other than the grayscale ones as shown in your images. Also, these colorful watermarks contain both the logo and text (mainly logo) other than the grayscale ones in yours. 

Hope it helps.

**Comment by Tristacheng** [comment]
Got it, thank you for your kindly reply. I will try it with my own dataset.

**Comment by Tristacheng** [comment]
> Got it, thank you for your kindly reply. I will try it with my own dataset.

hi,  I made my own dataset which consists of  100 watemarks and 30k pictures and 80% of watermark is grayscale
and retrained the model, however the results is not pleasing. finally the best performance is PSNR:31.2890 and SSIM:0.9644
For example, this surrounding of watermark arise halation
Could you give me some advise?
![image](https://user-images.githubusercontent.com/12381965/121350247-018b9b00-c95d-11eb-9462-71dab5e57588.png)

**Comment by vinthony** [comment]
Hi thanks for this feedback, 

1. Are these watermarks transparent?
2. It looks like the watermark detection fails to get accurate results, maybe the globally skip-connections can be removed, thus, the second network will still learn to handle the edge automatically.
3. By the way, I think a gradient loss may be helpful.

**Comment by Tristacheng** [comment]
> Hi thanks for this feedback,
> 
> 1. Are these watermarks transparent?
> 2. It looks like the watermark detection fails to get accurate results, maybe the globally skip-connections can be removed, thus, the second network will still learn to handle the edge automatically.
> 3. By the way, I think a gradient loss may be helpful.

1. yes, part of data in my dataset are transparent. However, I check the mask and found there is compression, and this might have a negative effect, so I will revise it and retrain my model.
![image](https://user-images.githubusercontent.com/12381965/121496955-87ffb580-ca0d-11eb-9e2a-38cab8b06188.png)
2. I think skip-connections might help to get a accurate watermark detection result, however, I will take a chance.
3. It is a good idea! I will try it.

Thank you for your suggestions,  I appreciate it.

**Comment by vinthony** [comment]
ok, btw, in 2. the globally skip-connections mean the skip-connection between the input and the final target, which means when the detection in the first step is not good enough, the second network still has a chance to make it better automatically.
